### DISCUSSION TITLE
Running local experiments

### DISCUSSION THREAD

**Discussion by markusgermar** [ctreffe/alfred#173] [General]
I would like to run my current experiment locally on laptops (not via mortimer). What is the best way to set-up the laptops (what should be installed etc...). How do the experimenters start the experiment? In short: What are your recommendations, warnings, ... when running local exepriments?

Thank you!

**Comment by ctreffe** [comment]
Hey Markus,

in order to run an experiment locally, you will need a Python environment with alfred3 installed. My only other recommendation is to use a current version of Google Chrome as web browser to run the experiments. Using Chrome will allow alfred to automatically open a browser window and load the experiment. There are no other prerequisites.

Starting an experiment is described on the [alfred3 documentation's landing page](https://alfredo3.psych.bio.uni-goettingen.de/docs/) and some additional information is provided [here](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.experiment.Experiment.run.html?highlight=run%20py#alfred3.experiment.Experiment.run). On a Windows machine, you should also be able to execute the experiment with a double click on your script.py file in the Windows explorer (however, this requires your Python 3 environment with alfred3 to be added to the Windows PATH, which might not be the case if you use multiple Python Environments or if you installed alfred3 into a virtual environment).

I don't have any warnings for you, but maybe a little tip regarding the experiment data: At the end of each experiment session on a local computer, alfred will automatically generate codebooks and datasets from all sessions stored on that machine. Instead of manually merging the files from each machine, you can just copy all saved sessions (the .json files in the save folder) to one machine and either run the experiment or use alfred's [command line interface](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.cli.html) to generate the final dataset as a .csv file.
